backend/infrahub/core/checks/tasks.py

```diff
@@ -0,0 +1,222 @@
+    impacted_artifacts = model.branch_diff.get_subscribers_ids(kind=InfrahubKind.ARTIFACT)
+
+    checks = []
+
+    for relationship in group.members.peers:
+        member = relationship.peer
+        artifact_id = artifacts_by_member.get(member.id)
+        if _render_artifact(
+            artifact_id=artifact_id,
+            managed_branch=model.source_branch_sync_with_git,
+            impacted_artifacts=impacted_artifacts,
+        ):
+            log.info(f"Trigger Artifact processing for {member.display_label}")
+
+            check_model = CheckArtifactCreate(
+                artifact_name=model.artifact_definition.artifact_name,
+                artifact_id=artifact_id,
+                artifact_definition=model.artifact_definition.definition_id,
+                commit=repository.source_commit,
+                content_type=model.artifact_definition.content_type,
+                transform_type=model.artifact_definition.transform_kind,
+                transform_location=model.artifact_definition.transform_location,
+                repository_id=repository.repository_id,
+                repository_name=repository.repository_name,
+                repository_kind=repository.kind,
+                branch_name=model.source_branch,
+                query=model.artifact_definition.query_name,
+                variables=member.extract(params=artifact_definition.parameters.value),
+                target_id=member.id,
+                target_name=member.display_label,
+                timeout=model.artifact_definition.timeout,
+                validator_id=validator.id,
+            )
+
+            checks.append(
+                service.workflow.submit_workflow(
+                    workflow=GIT_REPOSITORIES_CHECK_ARTIFACT_CREATE, parameters={"model": check_model}
+                )
+            )
+
+    await run_checks_and_update_validator(checks, validator)
+
+
+def _render_artifact(artifact_id: Optional[str], managed_branch: bool, impacted_artifacts: list[str]) -> bool:  # noqa: ARG001
+    """Returns a boolean to indicate if an artifact should be generated or not.
+    Will return true if:
+        * The artifact_id wasn't set which could be that it's a new object that doesn't have a previous artifact
+        * The source brance is not data only which would indicate that it could contain updates in git to the transform
+        * The artifact_id exists in the impacted_artifacts list
+    Will return false if:
+        * The source branch is a data only branch and the artifact_id exists and is not in the impacted list
+    """
+
+    # if not artifact_id or managed_branch:
+    #    return True
+    # return artifact_id in impacted_artifacts
+    # Temporary workaround tracked in https://github.com/opsmill/infrahub/issues/4991
+    return True
+
+
+@flow(name="git-repository-check-artifact-create", flow_run_name="Check artifact creation")
+async def create(model: CheckArtifactCreate, service: InfrahubServices) -> str:
```

Comment:
A minor comment here, I think that the organization might be a bit confusing if we refer to this as `infrahub.core.checks.tasks.create` just looking at this namespace it's hard to figure out that it's related to artifacts / artifact definitions. Could also be that we'd see more "conflicts" as we start to migrate other tasks. Potentially it would make more sense to have a root level ./artifacts directory.